streamlit_app.py

```diff
@@ -1,38 +1,55 @@
-from collections import namedtuple
-import altair as alt
-import math
-import pandas as pd
+import io
 import streamlit as st
-
-"""
-# Welcome to Streamlit!
-
-Edit `/streamlit_app.py` to customize this app to your heart's desire :heart:
-
-If you have any questions, checkout our [documentation](https://docs.streamlit.io) and [community
-forums](https://discuss.streamlit.io).
-
-In the meantime, below is an example of what you can do with just a few lines of code:
-"""
-
-
-with st.echo(code_location='below'):
-    total_points = st.slider("Number of points in spiral", 1, 5000, 2000)
-    num_turns = st.slider("Number of turns in spiral", 1, 100, 9)
-
-    Point = namedtuple('Point', 'x y')
-    data = []
-
-    points_per_turn = total_points / num_turns
-
-    for curr_point_num in range(total_points):
-        curr_turn, i = divmod(curr_point_num, points_per_turn)
-        angle = (curr_turn + 1) * 2 * math.pi * i / points_per_turn
-        radius = curr_point_num / total_points
-        x = radius * math.cos(angle)
-        y = radius * math.sin(angle)
-        data.append(Point(x, y))
-
-    st.altair_chart(alt.Chart(pd.DataFrame(data), height=500, width=500)
-        .mark_circle(color='#0068c9', opacity=0.5)
-        .encode(x='x:Q', y='y:Q'))
+from PIL import Image
+import numpy as np
+from tensorflow.keras.applications import EfficientNetB0
+from tensorflow.keras.preprocessing import image
+from tensorflow.keras.applications.efficientnet import preprocess_input, decode_predictions
+
+
+@st.cache(allow_output_mutation=True)
+def load_model():
+    return EfficientNetB0(weights="imagenet")
+
+
+def main():
+
+    st.title("Image classifications")
+    st.markdown("## Image recognition - using `tensorflow`, `streamlit` ")
+
+    def preprocess_image(img):
+        img = img.resize((224, 224))
+        num = image.img_to_array(img)
+        num = np.expand_dims(num, axis=0)
+        num = preprocess_input(num)
+        return num
+
+    def load_image():
+        uploaded_file = st.file_uploader(label="Choose an image for recognition")
+        if uploaded_file is not None:
+            image_data = uploaded_file.getvalue()
+            st.image(image_data)
+            return Image.open(io.BytesIO(image_data))
+        else:
+            return None
+
+    def print_predictions(pred):
+        classes = decode_predictions(pred, top=3)[0]
+        for cl in classes:
+            st.write(cl[1], cl[2])
```

Comment:
What is the reason you define these functions inside the `main` function? They don't use any scope information, so they can be defined as standalone functions.

Reply:
done